### ISSUE TITLE
/usr/bin/fwupdtool refresh SEGV

### ISSUE THREAD

**Issue by joakim-tjernlund** [fwupd/fwupd#2378]
**Describe the bug**
fwupdtool refresh SEGV on master
This is on an older HP laptop  

**Steps to Reproduce**
See above

**fwupd device information**
Please provide the output of the fwupd devices recognized in your system.

```shell
$ fwupdmgr get-devices --show-all-devices
```
```
fwupdtool get-devices --show-all-devices
Loading…                 [-                                      ]
OUI quirk required, please see https://github.com/fwupd/fwupd/wiki/ATA-Disk:-OUI-Quirk-Required!
---
[DeviceInstanceId=OUI\00080d]
Vendor = Toshiba
VendorId = ATA:0x1179
---
Loading…                 [***************************************]
HP EliteBook Folio 1040 G2
│
├─HD Graphics 5500:
│     Device ID:          bbbf1ce3d1cf15550c3760b354592040292415bb
│     Current version:    09
│     Vendor:             Intel Corporation (PCI:0x8086)
│     GUIDs:              19e1f63e-bef0-5582-b236-364069bcd0d5 ← PCI\VEN_8086&DEV_1616&REV_09
│                         64c0ea55-d861-5dd0-aa6b-bcc307f57ed4 ← PCI\VEN_8086&DEV_1616
│     Device Flags:       • Internal device
│                         • Cryptographic hash verification is available
│   
├─Intel(R) Core™ i7-5600U CPU @ 2.60GHz:
│     Device ID:          4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│     Current version:    0x0000002f
│     Vendor:             Intel
│     GUIDs:              b9a2dd81-159e-5537-a7db-e7101d164d3f ← cpu
│                         30249f37-d140-5d3e-9319-186b1bd5cac3 ← CPUID\PRO_0&FAM_06
│                         bcad68bd-c33a-56cd-a8a6-b9a78d62ee6e ← CPUID\PRO_0&FAM_06&MOD_3D
│                         a702f59f-10c2-5561-976c-b94c89cb42d8 ← CPUID\PRO_0&FAM_06&MOD_3D&STP_4
│     Device Flags:       • Internal device
│   
├─System Firmware:
│     Device ID:          e24dbdf2c162e9342bccb3d8114b2d2fbbd7a5ca
│     Current version:    M76 Ver. 01.16
│     Vendor:             Hewlett-Packard (DMI:Hewlett-Packard)
│     Update Error:       UEFI Capsule updates not available or enabled
│     GUID:               230c8b18-8d9b-53ec-838b-6cfc0383493a ← main-system-firmware
│     Device Flags:       • Internal device
│   
└─THNSFJ256GDNU A:
      Device ID:          7466579651466dd8b56d7a19479bed768440026d
      Summary:            ATA Drive
      Current version:    JYLA1102
      Vendor:             Toshiba (ATA:0x1179)
      Serial Number:      17BS11G8TJ4V
      GUIDs:              f260d559-ffd1-56ce-88a6-fc38c53d8e9d ← IDE\TOSHIBA_THNSFJ256GDNU_A_________________JYLA1102
                          9bc9dfb3-2789-5a46-afff-739966fddfd9 ← IDE\0TOSHIBA_THNSFJ256GDNU_A_________________
                          9d768a22-7625-501d-bfa7-7155287899e0 ← TOSHIBA THNSFJ256GDNU A
      Device Flags:       • Internal device
                          • Updatable
                          • Requires AC power
                          • Needs a reboot after installation
                          • Device is usable for the duration of the update


```
BT:
```
Reading symbols from /usr/bin/fwupdtool...
(No debugging symbols found in /usr/bin/fwupdtool)
[New LWP 17957]
[New LWP 17958]
[New LWP 17960]
[New LWP 17961]
[New LWP 17959]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `fwupdtool refresh'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f54f262efcf in g_source_set_ready_time (source=0x7f54e40078c0, ready_time=0)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/gmain.c:1942
1942	/var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/gmain.c: No such file or directory.
[Current thread is 1 (Thread 0x7f54eeb44e80 (LWP 17957))]
(gdb) bt full
#0  0x00007f54f262efcf in g_source_set_ready_time (source=0x7f54e40078c0, ready_time=0)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/gmain.c:1942
        context = 0x55652ee90720
        __func__ = "g_source_set_ready_time"
#1  0x00007f54f27b50d6 in cancellable_source_cancelled (cancellable=<optimized out>, user_data=0x7f54e40078c0)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gcancellable.c:666
        source = 0x7f54e40078c0
#2  0x00007f54f271d0e6 in _g_closure_invoke_va
    (closure=0x7f54e4008760, return_value=0x0, instance=0x55652ee87760, args=0x7fffcd61f920, n_params=0, param_types=0x0)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gclosure.c:873
        marshal = 0x7f54f271eef0 <g_cclosure_marshal_VOID__VOIDv>
        marshal_data = 0x0
        in_marshal = 0
        real_closure = 0x7f54e4008740
        __func__ = "_g_closure_invoke_va"
#3  0x00007f54f273a6d0 in g_signal_emit_valist
    (instance=0x55652ee87760, signal_id=<optimized out>, detail=0, var_args=var_args@entry=0x7fffcd61f920)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gsignal.c:3407
        return_accu = <optimized out>
        accu = 
              {g_type = 0x0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        accumulator = 0x0
        emission = 
          {next = 0x0, instance = 0x55652ee87760, ihint = {signal_id = 2, detail = 0, run_type = G_SIGNAL_RUN_FIRST}, state = EMISSION_RUN, chain_type = 0x55652edae590 [GCancellable]}
        signal_id = 2
        instance_type = <optimized out>
        emission_return = 
              {g_type = 0x0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        rtype = 0x4 [void]
--Type <RET> for more, q to quit, c to continue without paging--
        static_scope = 0
        fastpath_handler = <optimized out>
        closure = <optimized out>
        run_type = <optimized out>
        hlist = <optimized out>
        l = <optimized out>
        fastpath = 1
        instance_and_params = <optimized out>
        signal_return_type = <optimized out>
        param_values = <optimized out>
        node = <optimized out>
        i = <optimized out>
        n_params = <optimized out>
        __func__ = "g_signal_emit_valist"
#4  0x00007f54f273acaf in g_signal_emit (instance=instance@entry=0x55652ee87760, signal_id=<optimized out>, detail=detail@entry=0)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gsignal.c:3554
        var_args = {{gp_offset = 24, fp_offset = 48, overflow_arg_area = 0x7fffcd61fa00, reg_save_area = 0x7fffcd61f940}}
#5  0x00007f54f27b574a in g_cancellable_cancel (cancellable=0x55652ee87760 [GCancellable])
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gcancellable.c:509
        priv = 0x55652ee87740
#6  0x00007f54f28734a3 in _g_dbus_worker_close (worker=0x55652ee90610, task=<optimized out>)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gdbusprivate.c:1721
#7  0x00007f54f2873515 in _g_dbus_worker_stop (worker=0x55652ee90610)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gdbusprivate.c:1742
#8  0x00007f54f285e29f in g_dbus_connection_dispose (object=0x55652ee89030 [GDBusConnection])
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gdbusconnection.c:613
        connection = 0x55652ee89030 [GDBusConnection]
        __func__ = "g_dbus_connection_dispose"
#9  0x00007f54f2721f03 in g_object_unref (_object=<optimized out>)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3461
        weak_locations = 0x55652ee3b380
        old_ref = <optimized out>
        __func__ = "g_object_unref"
        object = 0x55652ee89030 [GDBusConnection]
--Type <RET> for more, q to quit, c to continue without paging--
        __func__ = "g_object_unref"
#10 g_object_unref (_object=0x55652ee89030) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3391
        object = 0x55652ee89030 [GDBusConnection]
        __func__ = "g_object_unref"
#11 0x00007f54f286fda8 in g_dbus_proxy_finalize (object=0x55652edc9f60 [GDBusProxy])
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gio/gdbusproxy.c:201
        proxy = 0x55652edc9f60 [GDBusProxy]
        __func__ = "g_dbus_proxy_finalize"
#12 0x00007f54f2721f7d in g_object_unref (_object=<optimized out>)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3499
        weak_locations = <optimized out>
        old_ref = <optimized out>
        __func__ = "g_object_unref"
        object = 0x55652edc9f60 [GDBusProxy]
        __func__ = "g_object_unref"
#13 g_object_unref (_object=0x55652edc9f60) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3391
        object = 0x55652edc9f60 [GDBusProxy]
        __func__ = "g_object_unref"
#14 0x00007f54f2b923f1 in  () at /usr/lib64/libfwupdplugin.so.1
#15 0x00007f54f2721f7d in g_object_unref (_object=<optimized out>)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3499
        weak_locations = <optimized out>
        old_ref = <optimized out>
        __func__ = "g_object_unref"
        object = 0x55652ee66180 [FuPlugin]
        __func__ = "g_object_unref"
#16 g_object_unref (_object=0x55652ee66180) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3391
        object = 0x55652ee66180 [FuPlugin]
        __func__ = "g_object_unref"
#17 0x00007f54f261dba2 in g_hash_table_remove_all_nodes
    (hash_table=hash_table@entry=0x55652ed9e520, notify=notify@entry=1, destruction=destruction@entry=1)
    at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/ghash.c:706
        i = 45
        key = <optimized out>
--Type <RET> for more, q to quit, c to continue without paging--c
        value = <optimized out>
        old_size = <optimized out>
        old_keys = 0x55652ee6b1b0
        old_values = 0x55652ee6b3c0
        old_hashes = 0x55652ee6b0a0
        old_have_big_keys = 1
        old_have_big_values = 1
#18 0x00007f54f261ef6f in g_hash_table_remove_all_nodes (destruction=1, notify=1, hash_table=0x55652ed9e520) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/ghash.c:1461
        __func__ = "g_hash_table_unref"
#19 g_hash_table_unref (hash_table=0x55652ed9e520) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/glib/ghash.c:1461
        __func__ = "g_hash_table_unref"
#20 0x000055652d33b8de in  ()
#21 0x00007f54f2721f7d in g_object_unref (_object=<optimized out>) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3499
        weak_locations = <optimized out>
        old_ref = <optimized out>
        __func__ = "g_object_unref"
        object = 0x55652edaad20 [FuPluginList]
        __func__ = "g_object_unref"
#22 g_object_unref (_object=0x55652edaad20) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3391
        object = 0x55652edaad20 [FuPluginList]
        __func__ = "g_object_unref"
#23 0x000055652d329fb8 in  ()
#24 0x00007f54f2721f7d in g_object_unref (_object=<optimized out>) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3499
        weak_locations = <optimized out>
        old_ref = <optimized out>
        __func__ = "g_object_unref"
        object = 0x55652edb50c0 [FuEngine]
        __func__ = "g_object_unref"
#25 g_object_unref (_object=0x55652edb50c0) at /var/tmp/portage/dev-libs/glib-2.64.5/glib-2.64.5/gobject/gobject.c:3391
        object = 0x55652edb50c0 [FuEngine]
        __func__ = "g_object_unref"
#26 0x000055652d31ef6f in main ()

**Comment by hughsie** [comment]
I think this is fixed by https://github.com/fwupd/fwupd/commit/50ff31de5690a5fda0aae7d0113980b5ebd3e926 -- can you try master please.

**Comment by joakim-tjernlund** [comment]
> I think this is fixed by [50ff31d](https://github.com/fwupd/fwupd/commit/50ff31de5690a5fda0aae7d0113980b5ebd3e926) -- can you try master please.

I already had that one in my master but I rebuild again just to make sure and now it work!
Oh, well
